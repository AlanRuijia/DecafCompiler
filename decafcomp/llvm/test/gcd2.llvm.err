; ModuleID = 'GreatestCommonDivisor'

declare void @print_int(i32)

define i32 @main() {
entry:
  %z = alloca i32
  store i32 0, i32* %z
  %a = load i32, i32* %a
  %b = load i32, i32* %b
  %calltmp = call i32 @gcd(i32 %a, i32 %b)
  call void @print_int(i32 %calltmp)
  ret i32 0
}

define i1 @iszero(i32, i32) {
entry:
  %a = alloca i32
  %b = alloca i32
  store i32 %0, i32* %a
  store i32 %1, i32* %b
  %b1 = load i32, i32* %b
  %EQTmp = icmp eq i32 %b1, 0
  %IfCond = icmp eq i1 %EQTmp, true
  br i1 %IfCond, label %Then, label %Else

Then:                                             ; preds = %entry
  %a2 = load i32, i32* %a
  ret i32 %a2
  br label %IfCont

Else:                                             ; preds = %entry
  %b3 = load i32, i32* %b
  %a4 = load i32, i32* %a
  %b5 = load i32, i32* %b
  %RemainTmp = srem i32 %a4, %b5
  %calltmp = call i32 @gcd(i32 %b3, i32 %RemainTmp)
  ret i32 %calltmp
  br label %IfCont

IfCont:                                           ; preds = %Else, %Then
  ret i1 true
}

define i32 @gcd(i32, i32) {
entry:
  %a = alloca i32
  %b = alloca i32
  store i32 %0, i32* %a
  store i32 %1, i32* %b
  %a1 = load i32, i32* %a
  %b2 = load i32, i32* %b
  %calltmp = call i1 @iszero(i32 %a1, i32 %b2)
  ret i1 %calltmp
  ret i32 0
}
