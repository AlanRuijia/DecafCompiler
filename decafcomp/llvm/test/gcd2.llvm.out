Debug message: Generating extern function...
Arg Type: 17 
Inserting method decl into symbol table...
Inserting block into symbol table...
Inserting method decl into symbol table...
Inserting block into symbol table...
Inserting method decl into symbol table...
Inserting block into symbol table...
Generating program...
Generating package...
Serching table...
FetchedObject: Found a
Serching table...
FetchedObject: Found b

Symbol Table: z 
Symbol Table: a b gcd iszero main 
Symbol Table: print_int 
Debug message: Generating MethodDecl...
Method Block CodeGen...
Block CodeGen......
Serching table...
FetchedObject: Found z
Allocating z
Debug message: Generating method call... Calling print_int
ArgumentListSize = 1
Argument Type: Debug message: Generating method call... Calling gcd
ArgumentListSize = 2
Argument Type: Serching table...
Table not found...
Serching table...
FetchedObject: Found a
(finishing codegen for this argument) int type
Func Arg: int type
Argument Type: Serching table...
Table not found...
Serching table...
FetchedObject: Found b
(finishing codegen for this argument) int type
Func Arg: int type
Method Call done...
(finishing codegen for this argument) int type
Func Arg: int type
Method Call done...
Finished Block CodeGen......
Finish Method Block CodeGen...
Generating Dummy Return 0...Finished...
Finish Method CodeGen...

Symbol Table: a b 
Symbol Table: a b gcd iszero main 
Symbol Table: print_int 
Debug message: Generating MethodDecl...
In IDTypeStringSpecialAST
Allocating a
Serching table...
FetchedObject: Found a
In IDTypeStringSpecialAST
Allocating b
Serching table...
FetchedObject: Found b
Serching table...
FetchedObject: Found a
Serching table...
FetchedObject: Found b
Method Block CodeGen...
Block CodeGen......
Serching table...
FetchedObject: Found b
Starting NumberExpr Codegen
Finished NumberExpr Codegen
OpId 11
Preparing to do code generation for If block...
Block CodeGen......
Block CodeGen for VarDeclList......
Block CodeGen for StmtList......
Serching table...
Serching table...
FetchedObject: Found a
Finished Block CodeGen......
Finished code generation for If block...
Preparing to do code generation for Else block...
Block CodeGen......
Block CodeGen for VarDeclList......
Block CodeGen for StmtList......
Debug message: Generating method call... Calling gcd
ArgumentListSize = 2
Argument Type: Serching table...
Serching table...
FetchedObject: Found b
(finishing codegen for this argument) int type
Func Arg: int type
Argument Type: Serching table...
Serching table...
FetchedObject: Found a
Serching table...
Serching table...
FetchedObject: Found b
OpId 6
(finishing codegen for this argument) int type
Func Arg: int type
Method Call done...
Finished Block CodeGen......
Finished Block CodeGen......
Finish Method Block CodeGen...
Generating Dummy Return true...Finished...
Finish Method CodeGen...

Symbol Table: a b 
Symbol Table: a b gcd iszero main 
Symbol Table: print_int 
Debug message: Generating MethodDecl...
In IDTypeStringSpecialAST
Allocating a
Serching table...
FetchedObject: Found a
In IDTypeStringSpecialAST
Allocating b
Serching table...
FetchedObject: Found b
Serching table...
FetchedObject: Found a
Serching table...
FetchedObject: Found b
Method Block CodeGen...
Block CodeGen......
Debug message: Generating method call... Calling iszero
ArgumentListSize = 2
Argument Type: Serching table...
FetchedObject: Found a
(finishing codegen for this argument) int type
Func Arg: int type
Argument Type: Serching table...
FetchedObject: Found b
(finishing codegen for this argument) int type
Func Arg: int type
Method Call done...
Finished Block CodeGen......
Finish Method Block CodeGen...
Generating Dummy Return 0...Finished...
Finish Method CodeGen...

Symbol Table: a b gcd iszero main 
Symbol Table: print_int 
